deeposlandia/kerastrain.py

```diff
@@ -0,0 +1,226 @@
+                              "during training"))
+    parser.add_argument('-e', '--nb-epochs',
+                        type=int,
+                        default=0,
+                        help=("Number of training epochs (one epoch means "
+                              "scanning each training image once)"))
+    parser.add_argument('-l', '--learning-rate',
+                        nargs="+",
+                        default=[0.001, 300, 0.95],
+                        type=float,
+                        help=("List of learning rate components (starting LR, "
+                              "decay steps and decay rate)"))
+    parser.add_argument('-s', '--image-size',
+                        default=256,
+                        type=int,
+                        help=("Desired size of images (width = height)"))
+    return parser
+
+def add_training_arguments(parser):
+    """Add training-specific arguments from the command line
+
+    Parameters
+    ----------
+    parser : argparse.ArgumentParser
+        Input parser
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        Modified parser, with additional arguments
+    """
+    parser.add_argument('-ii', '--nb-testing-image',
+                        type=int,
+                        default=5000,
+                        help=("Number of training images"))
+    parser.add_argument('-it', '--nb-training-image',
+                        type=int,
+                        default=18000,
+                        help=("Number of training images"))
+    parser.add_argument('-iv', '--nb-validation-image',
+                        type=int,
+                        default=2000,
+                        help=("Number of validation images"))
+    return parser
+
+if __name__=='__main__':
+    """Main method: run a convolutional neural network using Keras API
+    """
+
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description=("Convolutional Neural Netw"
+                                                  "ork on street-scene images"))
+    parser = add_instance_arguments(parser)
+    parser = add_hyperparameters(parser)
+    parser = add_training_arguments(parser)
+    args = parser.parse_args()
+
+    # Data path and repository management
+    aggregate_value = "full" if not args.aggregate_label else "aggregated"
+    folders = utils.prepare_folders(args.datapath, args.dataset, aggregate_value,
+                                    args.image_size, args.model)
+
+    # Instance name (name + image size + network size + batch_size
+    # + aggregate? + dropout + learning_rate)
+    instance_args = [args.name, args.image_size, args.network, args.batch_size,
+                     aggregate_value, args.dropout, utils.list_to_str(args.learning_rate)]
+    instance_name = utils.list_to_str(instance_args, "_")
+
+    # Data gathering
+    if (os.path.isfile(folders["training_config"]) and os.path.isfile(folders["validation_config"])
```

Comment:
Does it imply that if you don't have (or don't want to have) a testing dataset, this won't work?

Reply:
Thanks. :)
Actually that was a real mess, we need to be more organized. I hope it is good enough!

Reply:
Good question. At this point, we need to have a testing dataset, that is true. I considered that we generally have such a dataset, either by gathering third-part data, or by generating our own dataset.

The point is that we need to have a testing data if we want to do inference with the trained model, and this inference step is done further in the module. A side-question could be: do we make predictions on a testing set with the trained model in *this* module?

Reply:
It's a good question. The answer can wait. Let's use the code as is. We'll think about this if it's a problem or not.